### ISSUE TITLE
Note for no rewrite

### ISSUE THREAD

**Issue by alt-romes** [alt-romes/minecraft-dhall#1]
I recently used this again. Loading Dhall is really slow, and going around via JSON is unnecessarily complex. We are also kind of doing a build-system-y thing but from scratch (so e.g. we don't have recompilation checking).

If I were going to do this "right" in a more sustainable way, I would:

* Use `shake` to create the files and do recompilation avoidance
* Use Haskell itself instead of `dhall` for generating Item information;
    * To use Haskell, the user would write an executable with a main function which prints out (using default `Show/Read`) the list of Haskell Items, or the list of Haskell Blocks.

In short, the build tool would interpret the Haskell executables that return lists of items/blocks/..., read them as Haskell data types, and use shake-style build rules to create the appropriate files.

However, the best option is to use the [Fabric Data Generation](https://fabricmc.net/wiki/tutorial:datagen_setup) API and solution, which is maintained properly and does what I want wrt generating JSON models and such.
